contracts/src/Chainvoice.sol

```diff
@@ -1,104 +1,190 @@
         uint256 invoiceId = invoices.length;
-        invoices.push(InvoiceDetails({
-            id: invoiceId,
-            from: msg.sender,
-            to: to,
-            amountDue: amountDue,
-            isPaid: false
-        }));
-
-        // Track the invoice IDs for both sender and receiver
+        invoices.push(
+            InvoiceDetails({
+                id: invoiceId,
+                from: msg.sender,
+                dueDate:_dueDate,
+                issueDate: _issueDate,
+                user: user,
+                to: to,
+                client: client,
+                amountDue: amountDue,
+                isPaid: false
+            })
+        );
+
+        for (uint256 i = 0; i < _items.length; i++) {
+            itemDatas[invoiceId].push(_items[i]);
+        }
+
         sentInvoices[msg.sender].push(invoiceId);
         receivedInvoices[to].push(invoiceId);
 
         emit InvoiceCreated(invoiceId, msg.sender, to, amountDue);
     }
-    /**
-     * @dev Pay an invoice request using Ether
-     * @param invoiceId The ID of the invoice to be paid
-     */
+
+    function usdToNativeCurrencyConversion() public view returns (uint256) {
+        (, int256 answer, , , ) = priceFeed.latestRoundData();
+        require(answer > 0, "Invalid price data");
+        uint256 price = uint256(answer);
+        uint256 ethPriceAdjusted = price * 1e10;
+        uint256 nativeAmount = (feeAmountInUSD * 1e18);
+
+        return nativeAmount / ethPriceAdjusted;
+    }
+
+    uint256 public accumulatedFees;
+
     function payInvoice(uint256 invoiceId) external payable {
         require(invoiceId < invoices.length, "Invalid invoice ID");
         InvoiceDetails storage invoice = invoices[invoiceId];
         require(msg.sender == invoice.to, "Not authorized to pay this invoice");
         require(!invoice.isPaid, "Invoice already paid");
-        require(msg.value == invoice.amountDue, "Incorrect payment amount");
 
-        // Transfer Ether to the sender
-        (bool success,) = payable(invoice.from).call{value: msg.value}("");
+        uint256 feeAmountInNativeCurrency = usdToNativeCurrencyConversion();
+        require(
+            msg.value >= invoice.amountDue + feeAmountInNativeCurrency,
+            "Payment must cover the invoice amount and fee"
+        );
+        uint256 amountToRecipient = msg.value - feeAmountInNativeCurrency;
+        (bool success, ) = payable(invoice.from).call{value: amountToRecipient}(
+            ""
+        );
         require(success, "Payment transfer failed");
+        accumulatedFees += feeAmountInNativeCurrency;
         invoice.isPaid = true;
-        emit InvoicePaid(invoiceId, invoice.from, msg.sender, msg.value);
     }
-    /**
-     * @dev Get all invoices sent by the caller
-     * @return An array of InvoiceDetails for invoices sent by the caller
-     */
-    function getMySentInvoices() external view returns (InvoiceDetails[] memory) {
+
+    function getMySentInvoices()
```

Comment:
rename to `getSentInvoices` and take an address as an argument. Then this function will useful for any address, and not only for the msg.sender.

Also, note that view functions can be called without making a transaction and hence without having well-defined msg.sender.